src/openrct2/ride/RideData.h

```diff
@@ -175,6 +177,110 @@ enum class RideConstructionWindowContext : uint8_t
     Maze,
 };
 
+enum class RideTypeFlags : uint64_t
+{
+    HasTrackColourMain = 0,
+    HasTrackColourAdditional,
+    HasTrackColourSupports,
+    HasSinglePieceStation, // Set by flat rides, tower rides and shops/stalls.
+    HasLeaveWhenAnotherVehicleArrivesAtStation,
+    CanSynchroniseAdjacentStations,
+    TrackMustBeOnWater, // used only by boat Hire and submarine ride
+    HasGForces,
+    CannotHaveGaps, // used by rides that can't have gaps, like those with a vertical tower, such
+                    // as the observation tower
+    HasDataLogging,
+    HasDrops,
+    NoTestMode,
+    TrackElementsHaveTwoVarieties, // used by rides with two varieties, like the u and o
+                                   // shapes of the dinghy slide and the dry and submerged
+                                   // track of the water coaster
+    NoVehicles,                    // used only by maze, spiral slide and shops
+    HasLoadOptions,
+    HasNoTrack,
+    VehicleIsIntegral, // Set by flat rides where the vehicle is integral to the structure, like
+    // Merry-go-round and swinging ships. (Contrast with rides like dodgems.)
+    IsShopOrFacility,
+    TrackNoWalls, // if set, wall scenery can not share a tile with the ride's track
+    FlatRide,
+    PeepWillRideAgain, // whether or not guests will go on the ride again if they liked it
+                       // (this is
+    // usually applied to everything apart from transport rides)
+    PeepShouldGoInsideFacility, // used by toilets and first aid to mark that peep should go
+                                // inside the building (rather than 'buying' at the counter)
+    InRide,                     // peeps are "IN" (ride) rather than "ON" (ride)
+    SellsFood,
+    SellsDrinks,
+    IsToilet,
+    HasVehicleColours, // whether or not vehicle colours can be set
+    CheckForStalling,
+    HasTrack,
+    AllowExtraTowerBases, // Only set by lift
+    HasLargeCurves,       // whether the ride supports large (45 degree turn) curves
+    SupportsMultipleTrackColour,
+
+    AllowDoorsOnTrack,
+    MusicOnDefault,
+    AllowMusic,
+    HasAlternativeTrackType, // Used by the Flying RC, Lay-down RC, Multi-dimension RC
+    PeepCheckGForces,
+    HasEntranceExit,
+    AllowMoreVehiclesThanStationFits,
+    HasAirTime,
+    SingleSession,
+    AllowMultipleCircuits,
+    AllowCableLiftHill,
+    ShowInTrackDesigner,
+    TransportRide,
+    InterestingToLookAt,
+    SlightlyInterestingToLookAt,
+    StartConstructionInverted, // This is only set on the Flying RC and its alternative type.
+
+    ListVehiclesSeparately,
+    SupportsLevelCrossings,
+    IsSuspended,
+    HasLandscapeDoors,
+    UpInclineRequiresLift,
+    PeepCanUseUmbrella,
+    IsCashMachine,
+    HasOneStation,
+    HasSeatRotation,
+    IsFirstAid,
+    IsMaze,
+    IsSpiralSlide,
+
+    Count,
+};
+
+class RideTypeFlagsBitSet : public OpenRCT2::BitSet<static_cast<size_t>(RideTypeFlags::Count)>
```

Comment:
What is the benefit of extending bitset versus `using`?
```cpp
using RideTypeFlag= OpenRCT2::BitSet<RideTypeFlags::Count>;
```

Reply:
You would have to use static_cast for all the meta data. This way you use the static_cast from the extended constructor.

Reply:
This class is not supposed to be extended. It will be overall simplified just storing RideTypeFlags and having the operators implemented.